docs/DeveloperGuide.md

```diff
@@ -1,11 +1,308 @@
+1. When the user starts the program, it will first call `init()` to initialise the program. 
+2. A `StorageManager` object will be instantiated with the reference to a `ProjectManager` object that is used during load and save operations.
+3. `load()` will read the data file from the persistent storage, deserialise it into a `JsonObject` object and attempt to convert the object into its respective types.  
+  
+The program will exit immediately with an **exit code 1** if any of the conditions are met:
+- Error trying to read the file.
+- Conversion error due to missing properties.
+- Mapping error due to invalid property type (e.g. "name" is expecting a `String` but data read is an `Integer`).
+
+##### Converting and Mapping of JSON to Objects
+Due to the limitations of the library, parsing of the JSON string only converts it into either JsonObject or JsonArray objects which requires additional operations to map the data back to the respective model classes.  
+  
+As explained in [Storage Component](#storage-component), each model class except for `Priority` will inherit either `JsonableObject` or `JsonableArray` which are custom interfaces inheriting the `Jsonable` interface of _**json.simple**_. This requires the classes to implement the methods `toJson()` and `fromJson()`. This section will focus on `fromJson()`, which is used to implement the logic for **converting and mapping of JSON to objects of their respective type**.  
+  
+1. After loading the raw JSON string, `StorageManager` will call `Jsoner.deserialize()` to convert it into an object of `JsonObject`.
+2. At this point, `JsonObject` contains all the properties of `ProjectManager` and needs to be mapped back to a `ProjectManager` object.
+3. `StorageManager` will call `ProjectManager.fromJson()`, passing the `JSONObject` object from _Step 2_ as the parameter. 
+4. In `fromJson()`, there are two type of actions depending on the property it is mapping:  
+   1. **Type Casting**: Properties that are primitive or standard can be mapped directly through type casting (e.g. `int`, `boolean`, `String` etc.)
+   2. **\*Calling `fromJson()` of the Actual Type**: Any property that is originally a type of the **Scrumptious model classes** will be mapped with the following steps:
+      1. Create an empty object of the respective type (e.g. `Sprint`).
+      2. Type cast the property as `JsonObject` or `JsonArray` depending on the actual type of the property (e.g. `(JsonArray) object`).
+      3. Call `fromJson()` of the newly created object, passing the property as the parameter (e.g. `Sprint.fromJson()`).
+      4. New object's `fromJson()` will **repeat the same process again under Step 4** for its own properties.  
+      
+*`Priority` is an **enum** and is the only model which does not follow this strictly. It is mapped by type casting the property as `String` first, then calling the `Priority.valueOf()` method to convert it into its respective **enum**.  
```

Comment:
Perhaps you can link the definition of Priority under Glossary instead for better clarity